components/eam/src/physics/cam/check_energy.F90

```diff
@@ -1146,28 +1146,24 @@ subroutine energy_helper_eam_def(u,v,T,q,ps,pdel,phis, &
     integer, intent(in) :: ncol                   
     integer :: i,k                            
 
-    if (icldliq > 1  .and.  icldice > 1 .and. irain > 1 .and. isnow > 1) then
+    !if (icldliq > 1  .and.  icldice > 1 .and. irain > 1 .and. isnow > 1) then
        do i = 1, ncol
           call energy_helper_eam_def_column(u(i,:),v(i,:),T(i,:),q(i,1:pver,1:pcnst),&
                                    ps(i),pdel(i,:),phis(i), &
                                    ke(i),se(i),wv(i),wl(i),wi(i),wr(i),ws(i),te(i),tw(i) )                             
+          ! In the case where the micro-physics scheme is P3 there is no snow
+          ! constituent and isnow = -1.  So we still calculate the rest of the
+          ! consituents and ensure that ws = 0.0.  NOTE! This change will likely
+          ! lead to conflicts with upstream E3SM any time check_energy is
+          ! changed...  The most important thing is to avoid any SNOW calculations
+          ! when P3 is the microphysics scheme.
+          if (microp_scheme == 'P3') then
```

Comment:
Looks much better, not to be nitpicky but we can probably move this conditional to down below with the P3 check.  Then this chunk of code won't be too out of sync with upstream master.